src/mongocxx/uri.cpp

```diff
@@ -48,9 +48,12 @@ uri::uri(std::unique_ptr<impl>&& implementation) {
     _impl.reset(implementation.release());
 }
 
-uri::uri(bsoncxx::string::view_or_value uri_string)
-    : _impl(stdx::make_unique<impl>(libmongoc::uri_new(uri_string.terminated().data()))) {
-    if (_impl->uri_t == nullptr) {
+uri::uri(bsoncxx::string::view_or_value uri_string) {
+    bson_error_t error = {0};
```

Comment:
I think this is causing the ASAN issues on evergreen. Since `bson_error_t` is a _C_ type and not a _C++_ type it doesn't have a constructor. In general, _C_ isn't an object-oriented language. That means no constructors among other things. You could probably just remove the assignment here since `libmongoc::uri_new_with_error` will handle initialization.